src/txn/txn.c

```diff
@@ -381,7 +396,8 @@ __wt_txn_release(WT_SESSION_IMPL *session)
 	 */
 	__wt_txn_release_snapshot(session);
 	txn->isolation = session->isolation;
-	F_CLR(txn, WT_TXN_ERROR | WT_TXN_HAS_ID | WT_TXN_RUNNING);
+	F_CLR(txn, WT_TXN_ERROR | WT_TXN_HAS_ID |
+	    WT_TXN_NAMED_SNAPSHOT | WT_TXN_READONLY | WT_TXN_RUNNING);
```

Comment:
@keithbostic I thought it was ```WT_TXN_AUTOCOMMIT``` that was important to retain - since I believe it's managed at the API boundary. I'd like to get rid of this specific clearing of flags - it bit me a couple of times whilst doing the work.